scripts/generate-release-notes.js

```
@@ -0,0 +1,140 @@
+
+const NAME_RE = /^[a-zA-Z0-9_-]+$/;
+const SEMVER_RE = /^\d+\.\d+\.\d+$/;
+
+// --- Atomic write helper ---
+function atomicWrite(filePath, content) {
+  const tmp = filePath + '.tmp';
+  writeFileSync(tmp, content, 'utf8');
+  try {
+    renameSync(tmp, filePath);
+  } catch (e) {
+    try { unlinkSync(tmp); } catch (_) {}
```

ESLint

[error] Empty block statement.

(no-empty)